flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/CorrelateSortToRankRule.java

```diff
@@ -0,0 +1,256 @@
+                        .project(projects)
+                        .build();
+
+        call.transformTo(newRel);
+    }
+
+    /** Rule configuration. */
+    @Value.Immutable(singleton = false)
+    public interface CorrelateSortToRankRuleConfig extends RelRule.Config {
+        CorrelateSortToRankRule.CorrelateSortToRankRuleConfig DEFAULT =
+                ImmutableCorrelateSortToRankRule.CorrelateSortToRankRuleConfig.builder()
+                        .operandSupplier(
+                                b0 ->
+                                        b0.operand(Correlate.class)
+                                                .inputs(
+                                                        b1 ->
+                                                                b1.operand(Aggregate.class)
+                                                                        .oneInput(
+                                                                                b2 ->
+                                                                                        b2.operand(
+                                                                                                        Project
+                                                                                                                .class)
+                                                                                                .anyInputs()),
+                                                        b2 ->
+                                                                b2.operand(Sort.class)
+                                                                        .inputs(
+                                                                                b3 ->
+                                                                                        b3.operand(
+                                                                                                        Project
+                                                                                                                .class)
+                                                                                                .inputs(
+                                                                                                        b4 ->
+                                                                                                                b4.operand(
+                                                                                                                                Filter
+                                                                                                                                        .class)
+                                                                                                                        .anyInputs()))))
+                        .relBuilderFactory(FlinkRelFactories.FLINK_REL_BUILDER())
+                        .description("FlinkLimit0RemoveRule")
```

Comment:
```suggestion
                        .description("CorrelateSortToRankRule")
```

Reply:
thanks, good catch
done

I've also rebased to be sure it still passes based on most recent master